src/scripts/main.js

```diff
@@ -1,3 +1,66 @@
 'use strict';
 
-// write code here
+const appendRow = document.querySelector('.append-row');
+const removeRow = document.querySelector('.remove-row');
+const appendColumn = document.querySelector('.append-column');
+const removeColumn = document.querySelector('.remove-column');
+const field = document.querySelector('.field');
+const countOfRows = field.rows;
+
+function updateButtonStates() {
+  const rows = field.rows.length;
+  const countOFCells = field.rows[0].cells.length;
+
+  removeRow.disabled = rows <= 2;
+  appendRow.disabled = rows >= 10;
+  appendColumn.disabled = countOFCells >= 10;
+  removeColumn.disabled = countOFCells <= 2;
+}
+
+appendRow.addEventListener('click', (e) => {
+  if (e.target.classList.contains('append-row')) {
+    const newString = document.createElement('tr');
+    const countOfCell = document.querySelector('tr').children.length;
```

Comment:
The query `document.querySelector('tr')` assumes there is always at least one row in the table. If the table is empty, this will cause an error. Consider adding a check to ensure there is at least one row before accessing its children.